paimon-core/src/main/java/org/apache/paimon/schema/TableSchema.java

```diff
@@ -104,7 +104,8 @@ public TableSchema(
         this.highestFieldId = highestFieldId;
         this.partitionKeys = partitionKeys;
         this.primaryKeys = primaryKeys;
-        this.options = Collections.unmodifiableMap(options);
+        Objects.requireNonNull(options);
```

Comment:
Why need this? The @param `options` is not marked as @NotNull

Reply:
> Why need this? The @param `options` is not marked as @NotNull

removed